packages/geo/src/lib/map/shared/projection.utils.ts

```diff
@@ -66,3 +68,61 @@ export function computeProjectionsConstraints(
   };
   return projectionsConstraints;
 }
+
+export function detectFileEPSG(options: {
+  file: File;
+  epsgCode$: BehaviorSubject<string>;
+  nbLines?: number;
+}) {
+  const file = options.file;
+  const epsgCode$ = options.epsgCode$;
+  const nbLines = options.nbLines ?? 500;
+
+  if (
+    !file.name.toLowerCase().endsWith('.geojson') &&
+    !file.name.toLowerCase().endsWith('.json') &&
+    !file.name.toLowerCase().endsWith('.gml')
+  ) {
+    epsgCode$.next('epsgNotDefined');
+    return;
+  }
+  const reader = new FileReader();
+  reader.onload = (e) => {
```

Comment:
Présentement, on utilise un BehaviorSubject pour communiquer un output mais une alternative intéressante serait d'utiliser une Promesse ou un Observable pour wrapper le reader et controller la sortie.

```suggestion
export function detectFileEPSG(file: File, nbLines: number = 500): Promise<string> {
  // ... implementation
  const reader = new FileReader();
  reader.readAsText(file, 'UTF-8');

  return new Promise((resolve) => {
    reader.onload = (e) => {
      // ... implementation
      const espg = "...";
      resolve(epsg);
    }
  })
```

```
const epsg = await detectFileEPSG(file);
```

Reply:
fait.